### PR TITLE
chore(core): better diagnostics for table repair

### PR DESCRIPTION
**Pull request by puzpuzpuz**
Refs #3727

Existing logging is not enough to understand why table repair kicked in.

### CODE REVIEW

**Review by jerrinot — DISMISSED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 12 / 17 (70.59%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/pool/WriterPool.java|1|6|16.67%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/ApplyWal2TableJob.java|1|1|100.00%|

File: core/src/main/java/io/questdb/cairo/pool/WriterPool.java [hunk 1]
```diff
@@ -418,7 +418,7 @@ private TableWriter getWriterEntry(
             String lockReason,
```
Comment (ideoma) [inline core/src/main/java/io/questdb/cairo/pool/WriterPool.java:418]: nit:
```suggestion
            @NotNull String lockReason,
```

Comment (puzpuzpuz) [reply core/src/main/java/io/questdb/cairo/pool/WriterPool.java]: Done in 7adcc9e1dd6516e6450a730329ddf0e29003e5bc

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 15 / 21 (71.43%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/pool/WriterPool.java|0|5|00.00%|
|:large_blue_circle:|io/questdb/TelemetryConfigLogger.java|4|5|80.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/ApplyWal2TableJob.java|1|1|100.00%|

**Review by bluestreak01 — DISMISSED** [review]

File: core/src/main/java/io/questdb/cairo/CairoEngine.java [hunk 2]
```diff
@@ -1159,16 +1159,23 @@ private TableToken rename0(Path fromPath, TableToken fromTableToken, Path toPath
         }
     }
 
-    private void tryRepairTable(TableToken tableToken, RuntimeException rethrow) {
+    private void tryRepairTable(TableToken tableToken, CairoException rethrow) {
+        LOG.info()
```
[the remark this answers is not in the document]
Comment (puzpuzpuz) [reply core/src/main/java/io/questdb/cairo/CairoEngine.java:1163]: Do you think adding some logging to the `catch (EntryUnavailableException e)` block would help to make logging more complete?

[the remark this answers is not in the document]
Comment (bluestreak01) [reply core/src/main/java/io/questdb/cairo/CairoEngine.java:1163]: i think so, depending on how frequent these events are. Ideally looking at the log should not make us look at the code.

[the remark this answers is not in the document]
Comment (puzpuzpuz) [reply core/src/main/java/io/questdb/cairo/CairoEngine.java:1182]: Added more logging in b0c96b73282acd394c56d51813cdbe3103fc8d5a

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 14 / 16 (87.50%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/TelemetryConfigLogger.java|4|5|80.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|9|10|90.00%|
|:large_blue_circle:|io/questdb/cairo/wal/ApplyWal2TableJob.java|1|1|100.00%|

**Review by ideoma — APPROVED** [review]

**Pull request merged**